src/approaches/grammar_search_invention_approach.py

```diff
@@ -1097,13 +1097,21 @@ def _get_successors(
     init: FrozenSet[Predicate] = frozenset()
 
     # Greedy local hill climbing search.
-    path, _ = utils.run_hill_climbing(
-        init,
-        _check_goal,
-        _get_successors,
-        score_function.evaluate,
-        enforced_depth=CFG.grammar_search_hill_climbing_depth,
-        parallelize=CFG.grammar_search_parallelize_hill_climbing)
+    if CFG.grammar_search_search_algorithm == "hill_climbing":
+        path, _ = utils.run_hill_climbing(
+            init,
+            _check_goal,
+            _get_successors,
+            score_function.evaluate,
+            enforced_depth=CFG.grammar_search_hill_climbing_depth,
+            parallelize=CFG.grammar_search_parallelize_hill_climbing)
+    else:
+        assert CFG.grammar_search_search_algorithm == "gbfs"
```

Comment:
could we actually do this as an `elif` and have an `else` with an exception raised? that would make it easier to add new `elif` clauses in the future. you'll have to cover the `else` but that should be easy with a short test:
```
    # Test approach with unrecognized search algorithm.
    utils.update_config({
        "grammar_search_search_algorithm": "not a real search algorithm",
        "grammar_search_gbfs_num_evals": 10,
    })
    with pytest.raises(Exception) as e:
        _test_approach(env_name="cover",
                       approach_name="grammar_search_invention",
                       excluded_predicates="Holding",
                       try_solving=False,
                       sampler_learner="random")
    assert "Unrecognized grammar_search_search_algorithm" in str(e.value)
```

Reply:
done, thanks for writing the test :)

Reply:
i knew you wouldn't do it if i didn't ;)